integration-tests/smoke/keeper_test.go

```diff
@@ -377,6 +383,72 @@ func getKeeperSuite(
 					}, "1m", "1s").Should(Succeed())
 				})
 			}
+
+			if testToRun == RegisterUpkeepTest {
+				It("registers a new upkeep after the initial one was already registered and watches all of them perform", func() {
+					var initialCounters = make([]*big.Int, len(upkeepIDs))
+
+					// Observe that the upkeeps which are initially registered are performing and
+					// store the value of their initial counters in order to compare later on that the value increased.
+					Eventually(func(g Gomega) {
+						for i := 0; i < len(upkeepIDs); i++ {
+							counter, err := consumers[i].Counter(context.Background())
+							initialCounters[i] = counter
+							g.Expect(err).ShouldNot(HaveOccurred(), "Calling consumer's counter shouldn't fail")
+							g.Expect(counter.Int64()).Should(BeNumerically(">", int64(0)),
+								"Expected consumer counter to be greater than 0, but got %d", counter.Int64())
+							log.Info().Int64("Upkeep counter", counter.Int64()).Msg("Upkeeps performed")
+						}
+					}, "1m", "1s").Should(Succeed())
+
+					newConsumers, newUpkeepIDs := actions.RegisterNewUpkeeps(contractDeployer, networks, linkToken,
+						registry, registrar, upkeepGasLimit, 1)
+
+					// We know that newConsumers has size 1, so we can just use the newly registered upkeep.
+					newUpkeep := newConsumers[0]
+					newUpkeepID := newUpkeepIDs[0]
+
+					// Test that the newly registered upkeep is also performing.
+					Eventually(func(g Gomega) {
+						counter, err := newUpkeep.Counter(context.Background())
+						g.Expect(err).ShouldNot(HaveOccurred(), "Calling newly deployed upkeep's counter shouldn't fail")
+						g.Expect(counter.Int64()).Should(BeNumerically(">", int64(0)),
+							"Expected newly registered upkeep's counter to be greater than 0, but got %d", counter.Int64())
+						log.Info().Int64("Upkeep counter", counter.Int64()).Msg("Upkeeps performed")
+					}, "1m", "1s").Should(Succeed())
+
+					// While we registered the new upkeep, the initial upkeeps that were created first should have
+					// kept performing and hence their counters should be larger than what they were before.
+					for i := 0; i < len(upkeepIDs); i++ {
+						currentCounter, err := consumers[i].Counter(context.Background())
+						Expect(err).ShouldNot(HaveOccurred(), "Calling consumer's counter shouldn't fail")
+						log.Info().Int64("Upkeep counter", currentCounter.Int64()).Msg("Upkeep cancelled")
+
+						Expect(initialCounters[i].Int64() <= currentCounter.Int64()).To(BeTrue())
```

Comment:
Can we do a strict check here? (i.e. wait for them to be performed again)

Currently this doesn't assert much other than the fact that the counter can't reduce on contract

Reply:
Here I wanted to test that the number of times the upkeep executed has increased since the first time we performed the check (line 396). So I'm not sure what you mean by the counter can't reduce on contract?